overlord/state/taskrunner_test.go

```diff
@@ -125,19 +125,22 @@ var sequenceTests = []struct{ setup, result string }{{
 	result: "t11:do t11:do-error t12:do t12:do-block t12:do-unblock t12:do-retry",
 }, {
 	setup:  "t31:do-error t21:undo-error",
-	result: "t11:do t12:do t21:do t31:do t31:do-error t32:do t32:undo t21:undo t21:undo-error t11:undo",
+	result: "t11:do t12:do t21:do t22:do t31:do t31:do-error t32:do t32:undo t21:undo t21:undo-error t11:undo",
 }, {
 	setup:  "t21:do-set-ready",
-	result: "t11:do t12:do t21:do t31:do t32:do",
+	result: "t11:do t12:do t21:do t22:do t31:do t32:do t23:do",
 }, {
 	setup:  "t31:do-error t21:undo-set-ready",
-	result: "t11:do t12:do t21:do t31:do t31:do-error t32:do t32:undo t21:undo t11:undo",
+	result: "t11:do t12:do t21:do t22:do t31:do t31:do-error t32:do t32:undo t21:undo t11:undo",
 }, {
-	setup:  "t11:was-done:1 t12:was-done:2 t21:was-done:1,2 t31:was-done:1 t32:do-error:2",
+	setup:  "t11:was-done:1 t12:was-done:2 t21:was-done:1,2 t22:was-done:1,2 t31:was-done:1 t32:do-error:2",
 	result: "t31:undo t32:do t32:do-error t21:undo t11:undo",
 }, {
-	setup:  "t11:was-done:1 t12:was-done:2 t21:was-done:2 t31:was-done:2 t32:do-error:2",
+	setup:  "t11:was-done:1 t12:was-done:2 t21:was-done:2 t22:was-done:2 t31:was-done:2 t32:do-error:2",
 	result: "t31:undo t32:do t32:do-error t21:undo",
+}, {
+	setup:  "t11:was-done t12:was-done t21:was-done t22:was-done t31:was-done t32:was-done t23:do-error",
+	result: "t23:do t23:do-error t31:undo t32:undo t21:undo t11:undo",
```

Comment:
Why all the changes here?  I assume these tests were passing before since you didn't touch them earlier, if so please leave them alone to avoid breaking down the original cases they were intended to test.

Let's add a _new_ case, that checks specifically the change made. To make sure the change may be undone, and the new case here must break, then redo the fix.

Reply:
Done. The good news is the new test seems to be failing pretty consistently every time now without the bugfix in place.